static/js/message_list_view.js

```diff
@@ -132,6 +132,59 @@ function update_message_date_divider(opts) {
         timerender.render_date(curr_time, prev_time, today)[0].outerHTML;
 }
 
+function clear_unread_messages_divider(msg) {
+    msg.want_unread_divider = false;
+    msg.unread_divider_html = undefined;
+}
+
+function add_unread_messages_divider(msg) {
+    msg.want_unread_divider = true;
+    msg.include_sender = true;
+
+    var NEW_MESSAGES_TEXT = 'NEW MESSAGES';
+
+    if (msg.want_date_divider) {
+        msg.want_date_divider = false;
+        var unread_divider_html = $(msg.date_divider_html);
+        unread_divider_html.find('.date-line').after(
+            NEW_MESSAGES_TEXT,
+            '<hr class="date-line">'
+        );
+        var timerender_class = unread_divider_html.attr('class');
+        msg.unread_divider_html = '<span class="' + timerender_class + '">' +
+                                  unread_divider_html.html() +
+                                  '</span>';
+    } else {
+        msg.unread_divider_html = '<span>' + NEW_MESSAGES_TEXT + '</span>';
```

Comment:
Generally, building up HTML with `+` can be highly problematic for security.  It'd be better to do `$("span").text(NEW_MESSAGES_TEXT)` if we're constructing this with jQuery; but it might be even better to just use a very small handlebars template for this and pass in the details via the context.